android/src/main/java/nl/fhict/happynews/android/PostManager.java

```diff
@@ -32,19 +34,22 @@ private PostManager() {
      * @param c context of the apps
      * @return
      */
-    public ArrayList<Post> getNewPosts(Context c){
-        final ArrayList<Post> newPosts = new ArrayList<>();
+    public void updatePosts(Context c) {
+        newPosts = new ArrayList<>();
         Ion.with(c)
                 .load(API_URL)
-                .as(new TypeToken<List<Post>>(){})
+                .as(new TypeToken<List<Post>>() {
+                })
                 .setCallback(new FutureCallback<List<Post>>() {
                     @Override
                     public void onCompleted(Exception e, List<Post> posts) {
```

Comment:
We probably want to handle the case when the call fails. `Exception e` will be not null in that case.